src/CommunityToolkit.Aspire.Hosting.RavenDB/RavenDBServerSettings.cs

```diff
@@ -0,0 +1,136 @@
+
+/// <summary>
+/// Represents the settings for configuring a RavenDB server resource.
+/// </summary>
+public class RavenDBServerSettings
+{
+    /// <summary>
+    /// The iternal URL for the RavenDB server.
+    /// If not specified, the container resource will automatically assign a random URL.
+    /// </summary>
+    public string? ServerUrl { get; set; }
+
+    /// <summary>
+    /// The setup mode for the server. This determines whether the server is secured, uses Let's Encrypt, or is unsecured.
+    /// </summary>
+    public SetupMode SetupMode { get; set; }
+
+    /// <summary>
+    /// Gets the licensing options configured for the server.
+    /// </summary>
+    public LicensingOptions? LicensingOptions { get; private set; }
+
+    /// <summary>
+    /// Creates an unsecured RavenDB server settings object with default settings.
+    /// </summary>
+    public static RavenDBServerSettings Unsecured() => new RavenDBServerSettings { SetupMode = SetupMode.None };
+
+    /// <summary>
+    /// Creates a secured RavenDB server settings object with the specified configuration.
+    /// </summary>
+    /// <param name="setupMode">The setup mode to use. Must be <see cref="SetupMode.LetsEncrypt"/> or <see cref="SetupMode.Secured"/>.</param>
+    /// <param name="domainUrl">The public domain URL for the server.</param>
+    /// <param name="certificatePath">The path to the certificate file.</param>
+    /// <param name="certificatePassword">The password for the certificate file, if required. Optional.</param>
+    /// <param name="serverUrl">The optional server URL.</param>
+    public static RavenDBServerSettings Secured(SetupMode setupMode, string domainUrl, string certificatePath,
+        string? certificatePassword = null, string? serverUrl = null)
+    {
+        if (setupMode == SetupMode.None || setupMode == SetupMode.Unsecured)
+            throw new InvalidDataException(
+                $"Setup mode must be of type: '{SetupMode.LetsEncrypt}' or '{SetupMode.Secured}'");
+
+        return new RavenDBSecuredServerSettings
+        {
+            SetupMode = setupMode,
+            ServerUrl = serverUrl,
+            CertificatePath = certificatePath,
+            CertificatePassword = certificatePassword,
+            PublicServerUrl = domainUrl
+        };
+    }
+
+    /// <summary>
+    /// Configures licensing options for the RavenDB server.
+    /// </summary>
+    /// <param name="license">The license string for the RavenDB server.</param>
+    /// <param name="eulaAccepted">Indicates whether the End User License Agreement (EULA) has been accepted. Defaults to <c>true</c>.</param>
+    public void WithLicense(string license, bool eulaAccepted = true)
+    {
+        LicensingOptions = new LicensingOptions
+        {
+            EulaAccepted = eulaAccepted,
+            License = license
```

Comment:
I thought about this too. since supporting `LicensePath` would require a bind mount from the host to the container resource, I don’t think it should be part of `RavenDBServerSettings`. 
perhaps we could consider adding it as a parameter to the overloads, but even then, it starts to feel a bit much IMO. I guess we can document this scenario instead. what do you think?

Reply:
agree, we can add it later if needed